### DISCUSSION TITLE
Does some client version support NX and GET options together in the SET command?

### DISCUSSION THREAD

**Discussion by Andrey-Metelyov** [xetorthio/jedis#3921] [Q&A] [answered]
> Starting with Redis version 7.0.0: Allowed the NX and GET options to be used together.

So i can execute following commands:
```
>> set mykey myvalue nx get

(nil)

>> set mykey myvalue nx get

"myvalue"
```
If the key is successfully set, `nil` is returned, and if the key was already set previously, its value is returned.

So, question: does any version of redis client support `NX` and `GET` options together in a `SET` command?

Current version of method `public String set(final String key, final String value, final SetParams params)` supports only `NX|XX, EX|PX` params and returns `OK` or `null`.

**Comment by sazzad16** [comment, marked as answer]
There is `String setGet(String key, String value, SetParams params);`
